### PR TITLE
Use catalog-info.yaml for release metadata.  Get rid of openedx.yaml

### PR DESCRIPTION
**Pull request by feanil**
OEP-55:ADR-4 in this PR has most of the context of what I'm proposing so I suggest reading that first.  The rest is just a consequence of that decision.

- docs: Move OEP-2 to archived.
- docs: Change how we track release data.
- docs: Update documents as a consequence of OEP-55:ADR-4
- docs: Other minor docs cleanup.

### CODE REVIEW

File: oeps/processes/oep-0055/decisions/0004-release-data-in-catalog-info-files.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+******
+
+**Accepted**
+
+Context
+*******
+
+* Currently we have 2 high-level metadata files that will live in every repo. The ``openedx.yaml`` file and the ``catalog-info.yaml`` file.
+
+* The only information in ``openedx.yaml`` that is still useful and kept up-to-date is ``release`` tag that tells you what repos are part of the twice-annual Open edX release.
```
Comment (nedbat) [inline oeps/processes/oep-0055/decisions/0004-release-data-in-catalog-info-files.rst:14]:
```suggestion
* The only information in ``openedx.yaml`` that is still useful and kept up-to-date is the ``release`` tag that tells you what repos are part of the twice-annual Open edX release.
```

**Review by nedbat — APPROVED** [review]
One missing word, then merge it!

**Review by jalondonot — APPROVED** [review]
Seems good to me!

**Review by jalondonot — APPROVED** [review]

**Comment by feanil** [comment]
@nedbat good point, I've created https://github.com/openedx/open-edx-proposals/pull/529 to address that.

**Comment by kdmccormick** [comment]
@feanil could you toss an issue on the [BTR board ](https://github.com/orgs/openedx/projects/28) so that we remember to apply this change before Redwood?

**Comment by feanil** [comment]
@kdmccormick I think nothing needs to change on the BTR side, once https://github.com/openedx/repo-tools/pull/467 lands, it should to work automatically.

**Comment by kdmccormick** [comment]
Ah, I wasn't aware of that PR. Cool, thanks.

**Pull request merged**